src/shape/bounding-box.js

```diff
@@ -421,6 +421,12 @@ class BoundingBox {
         return sq;
     }
 
+    area() {
+        const extents = this.halfExtents.clone();
+        extents.mulScalar(2);
+        return extents.x * extents.y + extents.z * extents.x + extents.y * extents.z;
+    }
```

Comment:
No matter where this lands, it is a getter and can be shortened aswell with less lookups:

```suggestion
    get area() {
        const {x, y, z} = this.halfExtents.clone().mulScalar(2);
        return x * y + z * x + y * z;
    }
```

(the `.area()` braces need to be removed of course too)

Reply:
personally, I'd prefer solution without .clone() in it, as it allocates object. We only clone is so that we can multiply it by 2 .. I think we should just factor that multiplication into the last line directly.

Reply:
Updated to least amount of GC and float operations, can you counter-check?

Reply:
looks good to me.
Also, the box has 6 sides .. but we only add up areas of 3 of them .. to be really correct, the result should be *2 ?

Reply:
> Also, the box has 6 sides .. but we only add up areas of 3 of them .. to be really correct, the result should be *2 ?

Yes, I agree, maybe its not important for the algorithm, but @sri-30 has to check it now. Otherwise I guess it should be called `halfArea` or something

Reply:
My thoughts exactly; since it's only required for a heuristic. I'll rename it to halfArea, and use the version that @kungfooman has suggested.